dex/src/main/scala/com/wavesplatform/dex/actors/address/AddressActor.scala

```diff
@@ -341,26 +379,56 @@ class AddressActor(
     case query: Query.GetTradableBalance => sender() ! Reply.GetBalance(balances.tradableBalance(query.forAssets).filter(_._2 > 0))
 
     case Query.GetOrderStatus(orderId) =>
-      sender() ! Reply.GetOrderStatus(activeOrders.get(orderId).fold[OrderStatus](orderDB.status(orderId))(_.status))
+      val origSender = sender()
+      activeOrders.get(orderId) match {
+        case Some(order) =>
+          origSender ! Reply.GetOrderStatus(order.status)
+        case None =>
+          orderDb.status(orderId).onComplete {
+            case Success(x) =>
+              origSender ! Reply.GetOrderStatus(x)
+            case Failure(th) =>
+              log.error("error while retrieving order status", th)
+              origSender ! UnexpectedError
+          }
+      }
 
     case Query.GetOrderStatusInfo(orderId) =>
-      sender() ! Reply.GetOrdersStatusInfo(
-        activeOrders
-          .get(orderId)
-          .map(ao => OrderInfo.v6(ao, ao.status)) orElse orderDB.getOrderInfo(orderId)
-      )
+      val origSender = sender()
+      activeOrders.get(orderId).map(ao => OrderInfo.v6(ao, ao.status)) match {
+        case maybeOrderInfo @ Some(_) =>
+          origSender ! Reply.GetOrdersStatusInfo(maybeOrderInfo)
+        case None =>
+          orderDb.getOrderInfo(orderId).onComplete {
+            case Success(maybeOrderInfo) =>
+              origSender ! Reply.GetOrdersStatusInfo(maybeOrderInfo)
+            case Failure(th) =>
+              log.error("error while retrieving order info", th)
+              origSender ! UnexpectedError
+          }
+      }
 
     case Query.GetOrdersStatuses(maybePair, orderListType) =>
       val matchingActiveOrders =
-        if (orderListType.hasActive)
-          getActiveLimitOrders(maybePair)
-            .map(ao => ao.id -> OrderInfo.v6(ao, ao.status))
-            .toSeq
-            .sorted
-        else Seq.empty
-
-      val matchingClosedOrders = if (orderListType.hasClosed) orderDB.getFinalizedOrders(owner, maybePair) else Seq.empty
-      sender() ! Reply.GetOrderStatuses(matchingActiveOrders ++ matchingClosedOrders)
+        (if (orderListType.hasActive)
+           getActiveLimitOrders(maybePair)
+             .map(ao => ao.id -> OrderInfo.v6(ao, ao.status))
+             .toSeq
+             .sorted
+         else Seq.empty).toList
```

Comment:
Or just convert to `List` and sort. `sorted` is working with converting to `Array` under the hood for both `Vector` and `List`.